guides/source/action_controller_overview.md

```diff
@@ -1297,10 +1297,12 @@ While any newly generated Rails applications will have the health check at `/up`
 
 ```ruby
 Rails.application.routes.draw do
-  get "healthz" => "rails/health#show"
+  get "healthz" => "rails/health#show", as: :rails_health_check
 end
 ```
 
 The health check will now be accessible via the `/healthz` path.
 
-NOTE: This endpoint is not designed to give the status of all of your application's dependencies, such as the database or redis cluster. It is also not recommended to use those for health checks, in general, as it can lead to situations where your application is being restarted due to a third-party service going bad. Ideally, you should design your application to handle those outages gracefully.
\ No newline at end of file
+NOTE: This endpoint does not reflect the status of all of your service's dependencies, such as the database or redis cluster. Replace "rails/heath#show" with your own controller action if you have application specific needs.
```

Comment:
@rubys My only knit is that we should be using "application's dependencies" here, I know it's bikeshedding but we refer to them as Rails Apps everywhere so trying to be consistent. When I first wrote this part, I also used "service" coincidentally :joy:

I'm going to update this part, but appreciate you updating the docs here, it re-affirms my instincts about not putting too much cruft in a health check :)

Reply:
@zzak go for it.  Those weren't my words, so I didn't change them.

I'm afraid that we are going to need to agree to disagree about "putting too much cruft in a health check".  😄  Over the next few days I'll be writing up a blog post about health checks and fly.io.  I'll be putting everything in a positive light, but the net of it will be that we provide tcp checks (i.e. "is the app listening on the socket?") without any need for any app support.  Adding a http check (e.g., with a minimal "/up" endpoint) does add some value, but it is minimal.  Routing around regional failures require a bit more, and there things like easymon or rails-healthcheck come in handy.

When the post is ready, it will be posted here: https://fly.io/ruby-dispatch/

Reply:
And the promised post is finally live: https://fly.io/ruby-dispatch/health-checks/